src/Features/Core/Portable/CodeLens/DisplayInfoProvider.cs

```diff
@@ -0,0 +1,346 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CodeLens
+{
+    internal static class DisplayInfoProvider
+    {
+        private static bool IsValueParameter(ISymbol symbol)
+        {
+            if (symbol is IParameterSymbol)
+            {
+                var method = symbol.ContainingSymbol as IMethodSymbol;
+                if (method != null)
+                {
+                    if (method.MethodKind == MethodKind.EventAdd ||
+                        method.MethodKind == MethodKind.EventRemove ||
+                        method.MethodKind == MethodKind.PropertySet)
+                    {
+                        return symbol.Name == "value";
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private static Glyph? GetGlyph(ISymbol symbol)
+        {
+            while (true)
```

Comment:
I think Roslyn already have this code somewhere that navigate to or find all reference uses.